app/src/main/java/org/jellyfin/mobile/player/source/LocalJellyfinMediaSource.kt

```diff
@@ -0,0 +1,20 @@
+package org.jellyfin.mobile.player.source
+
+import kotlinx.serialization.Serializable
+import org.jellyfin.sdk.model.api.BaseItemDto
+import org.jellyfin.sdk.model.api.MediaSourceInfo
+import org.jellyfin.sdk.model.api.PlayMethod
+import java.util.UUID
+
+@Serializable(with = JellyfinMediaSourceSerializer::class)
+class LocalJellyfinMediaSource(
+    itemId: UUID,
+    item: BaseItemDto?,
+    sourceInfo: MediaSourceInfo,
+    playSessionId: String,
+    startTimeTicks: Long? = null,
+    audioStreamIndex: Int? = null,
+    subtitleStreamIndex: Int? = null,
+) : JellyfinMediaSource(itemId, item, sourceInfo, playSessionId, startTimeTicks, audioStreamIndex, subtitleStreamIndex) {
```

Comment:
I think it could be really nice adding the download length, download folder URI and download URL as paramters to the LocalJellyfinMediaSource, this would allow easier extension and simplify the Room database

Reply:
I like this! Plus it would help with using (/force us to use lol) the type system to our advantage and prevent all these casts everywhere!

Reply:
Resolve later